lib/fastly_nsq/message_queue/listener.rb

```diff
@@ -1,8 +1,10 @@
 module MessageQueue
   class Listener
-    def initialize(topic:, channel:)
-      @topic = topic
-      @channel = channel
+    def initialize(topic:, channel:, processor: nil, consumer: nil)
+      @topic     = topic
+      @channel   = channel
+      @processor = processor || DEFAULT_PROCESSOR
```

Comment:
That's true. Is it more readable/intention-revealing in that form? I think, long-term, I'd like to see us move away from the "magic constant" `MessageProcessor` in favor of this kind of configuration (maybe with some sane defaults), so if we need to establish how we're going to write that sort of thing now, maybe it's a good time.